sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/TDigestQuantiles.java

```diff
@@ -0,0 +1,345 @@
+ * </code></pre>
+ *
+ * <p><b>Warning: this class is experimental.</b> <br>
+ * Its API is subject to change in future versions of Beam.
+ * */
+@Experimental
+public final class TDigestQuantiles {
+
+  /**
+   * Compute the stream in order to build a T-Digest structure (MergingDigest)
+   * for keeping track of the stream distribution and returns a {@code PCollection<MergingDigest>}.
+   * <br> The resulting structure can be queried in order to retrieve the approximate value
+   * at a given quantile or the approximate quantile position of a given element.
+   */
+  public static GlobalDigest globally() {
+    return GlobalDigest.builder().build();
+  }
+
+  /**
+   * Like {@link #globally()}, but builds a digest for each key in the stream.
+   *
+   * @param <K>             the type of the keys
+   */
+  public static <K> PerKeyDigest<K> perKey() {
+    return PerKeyDigest.<K>builder().build();
+  }
+
+  /** Implementation of {@link #globally()}. */
+  @AutoValue
+  public abstract static class GlobalDigest
+          extends PTransform<PCollection<Double>, PCollection<MergingDigest>> {
+
+    abstract double compression();
+
+    abstract Builder toBuilder();
+
+    static Builder builder() {
+      return new AutoValue_TDigestQuantiles_GlobalDigest.Builder()
+              .setCompression(100);
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setCompression(double cf);
+
+      abstract GlobalDigest build();
+    }
+
+    /**
+     * Returns a new {@link PTransform} with a new compression factor {@code cf}.
+     *
+     * <p>This value bounds the maximum number of points resumed in each centroid
+     * and the total number of centroids in the digest.
+     *
+     * @param cf the bound value for centroid and digest sizes.
+     */
+    public GlobalDigest withCompression(double cf) {
+      return toBuilder().setCompression(cf).build();
+    }
+
+    @Override
+    public PCollection<MergingDigest> expand(PCollection<Double> input) {
+      return input.apply(
+                      "Compute T-Digest Structure",
+                      Combine.globally(TDigestQuantilesFn.create(this.compression())));
+    }
+  }
+
+  /** Implementation of {@link #perKey()}. */
+  @AutoValue
+  public abstract static class PerKeyDigest<K>
+          extends PTransform<PCollection<KV<K, Double>>, PCollection<KV<K, MergingDigest>>> {
+
+    abstract double compression();
+    abstract Builder<K> toBuilder();
+
+    static <K> Builder<K> builder() {
+      return new AutoValue_TDigestQuantiles_PerKeyDigest.Builder<K>()
+              .setCompression(100);
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder<K> {
+      abstract Builder<K> setCompression(double cf);
+
+      abstract PerKeyDigest<K> build();
+    }
+
+    /**
+     * Returns a new {@link PTransform} with a new compression factor {@code cf}.
+     *
+     * <p>This value bounds the maximum number of points resumed in each centroid
```

Comment:
Please refer to the class-level javadoc for advice on choosing a good value

Reply:
I said the same at class level